cdc/cdc.cc

```diff
@@ -218,4 +219,195 @@ db_context db_context::builder::build() {
     };
 }
 
+class transformer final {
+public:
+    struct map_cmp {
+        template <typename Func>
+        bool cmp(const std::pair<net::inet_address, unsigned int>& a,
+                 const std::pair<net::inet_address, unsigned int>& b,
+                 Func&& func) const {
+            auto&& a_ip = func(a.first).ip;
+            auto&& b_ip = func(b.first).ip;
+            return a_ip < b_ip || (a_ip == b_ip && a.second < b.second);
+        }
+        bool operator()(const std::pair<net::inet_address, unsigned int>& a,
+                        const std::pair<net::inet_address, unsigned int>& b) const {
+            if (a.first.is_ipv4()) {
+                return !b.first.is_ipv4() ||
+                    cmp(a, b, [] (const net::inet_address& addr) { return addr.as_ipv4_address(); });
+            } else {
+                return !b.first.is_ipv4() &&
+                    cmp(a, b, [] (const net::inet_address& addr) { return addr.as_ipv6_address(); });
+            }
+        }
+    };
+    using streams_type = std::map<std::pair<net::inet_address, unsigned int>, utils::UUID, map_cmp>;
+private:
+    db_context _ctx;
+    const schema& _schema;
+    schema_ptr _log_schema;
+    utils::UUID _time;
```

Comment:
How can the same time be valid for all transformations? Shouldn't it come from the mutation itself?

Reply:
It's the time the change reached the cluster.